ai-integrations/langgraph.ipynb

```diff
@@ -0,0 +1,598 @@
+    "   collection = collection,\n",
+    "   embedding = embedding_model,\n",
+    "   text_key = \"plot\",\n",
+    "   embedding_key = \"plot_embedding\",\n",
+    "   relevance_score_fn = \"dotProduct\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use helper method to create the vector search index\n",
+    "vector_store.create_vector_search_index(\n",
+    "   dimensions = 1536 # The dimensions of the vector embeddings to be indexed \n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_mongodb.index import create_fulltext_search_index\n",
+    "import time\n",
+    "\n",
+    "# Use helper method to create the search index\n",
+    "create_fulltext_search_index(\n",
+    "   collection = collection,\n",
+    "   field = \"title\",\n",
+    "   index_name = \"search_index\"\n",
+    ")\n",
+    "\n",
+    "# Wait for the index to build (this can take around a minute)\n",
+    "time.sleep(60)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "pZfheX5FiIhU"
+   },
+   "source": [
+    "## Define agent tools\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.agents import tool\n",
+    "\n",
+    "# Define a vector search tool\n",
+    "@tool\n",
+    "def vector_search(user_query: str) -> str:\n",
+    "    \"\"\"\n",
+    "    Retrieve information using vector search to answer a user query.\n",
+    "\n",
+    "    Args:\n",
+    "    user_query (str): The user's query string.\n",
+    "\n",
+    "    Returns:\n",
+    "    str: The retrieved information formatted as a string.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    retriever = vector_store.as_retriever(\n",
+    "       search_type = \"similarity\",\n",
+    "       search_kwargs = { \"k\": 5 } # Retrieve top 5 most similar documents\n",
+    "    )\n",
+    "\n",
+    "    results = retriever.invoke(user_query)\n",
+    "   \n",
+    "    # Concatenate the results into a string\n",
+    "    context = \"\\n\\n\".join([f\"{doc.metadata['title']}: {doc.page_content}\" for doc in results])\n",
+    "    return context\n",
+    "\n",
+    "# Test the tool\n",
+    "test_results = vector_search.invoke(\"What are some movies that take place in the ocean?\")\n",
+    "print(test_results)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_mongodb.retrievers.full_text_search import MongoDBAtlasFullTextSearchRetriever\n",
+    "\n",
+    "# Define a full-text search tool\n",
+    "@tool\n",
+    "def full_text_search(user_query: str) -> str:\n",
+    "    \"\"\"\n",
+    "    Retrieve movie plot content based on the provided title.\n",
```

Comment:
Same comment here - we haven't done args/returns in any of our other docs code examples so this stood out to me.